pkg/kubelet/server/metrics/metrics.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	kubeletSubsystem = "kubelet"
+)
+
+var (
+	// HTTPRequests tracks the number of the http requests received since the server started (excluding long-running requests)
+	HTTPRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_total",
+			Help:      "Number of the http requests received since the server started (excluding long-running requests).",
+		},
+		// server_type aims to differentiate the readonly server and the readwrite server.
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPRequestsDuration tracks the duration in seconds to serve http requests (excluding long-running requests)
+	HTTPRequestsDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_duration_seconds",
+			Help:      "Duration in seconds to serve http requests (excluding long-running requests).",
+			// Use DefBuckets for now, will customize the buckets if necessary.
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPInflightRequests tracks the number of the http requests received in last seconds (excluding long-running requests)
+	HTTPInflightRequests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_inflight_requests_gauge",
+			Help:      "Number of the http requests received in last seconds (excluding long-running requests).",
```

Comment:
Is this the number received in the last X seconds?  Or the number of requests that haven't been responded to yet?

Reply:
This is the number received in the last X seconds, which is set to 1 second currently in `StartRecordingInflightRequests`.

Reply:
I saw APIServer is capturing the rate with a similar metric, [currentInflightRequests](https://github.com/kubernetes/kubernetes/blob/559b11410ee51998316e3e80d864886d88314607/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L146), so I did this too. However, if deriving the rate from the request count is a better practice, I can definitely switch.

It seems that a separate metric is more accurate in that it can capture the highs and lows of traffic. 

Any suggestions?

Reply:
Updated how this metric is collected so that it tracks the inflight requests.